terminus/builders/procedural_city/vertex_graph_to_roads_converter.py

```diff
@@ -48,12 +54,21 @@ def _build_road(self, street, previous_vertex, current_vertex):
                 edges.append([neighbour,
                              abs(current_angle - neighbours_angle)])
             edges.sort(key=itemgetter(1))
+            # Pick the edge with the lowest angle and check if it's below the
+            # threshold.
             if edges and (edges[0][1] <= self.angle_threshold):
                 selected_neighbour = edges[0][0]
-                current_vertex.neighbours.remove(selected_neighbour)
-                selected_neighbour.neighbours.remove(current_vertex)
-                street.add_point(selected_neighbour.coords)
-                self._build_road(street, current_vertex, selected_neighbour)
+                # If we are connecting a street to either a street or a trunk
+                # it's ok to continue, also if the current and next vertices
+                # are part of a trunk.
```

Comment:
Just wondering: are you considering this case:

- You are building a street already.
- Your best candidate (e.g. 20deg) is a trunk.
- You next best candidate (e.g. 30deg) is a street.

In such a case you should continue building a street with the second best candidate.